viz/cmd/stat.go

```diff
@@ -167,8 +166,31 @@ If no resource name is specified, displays stats about all resources of the spec
 
   # Get all inbound stats to the test namespace.
   linkerd viz stat ns/test`,
-		Args:      cobra.MinimumNArgs(1),
-		ValidArgs: pkg.ValidTargets,
+		Args: cobra.MinimumNArgs(1),
+		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+
+			k8sAPI, err := k8s.NewAPI(kubeconfigPath, kubeContext, impersonate, impersonateGroup, 0)
+			if err != nil {
+				return nil, cobra.ShellCompDirectiveError
+			}
+
+			if options.namespace == "" {
+				options.namespace = pkgcmd.GetDefaultNamespace(kubeconfigPath, kubeContext)
+			}
+
+			cc := k8s.NewCommandCompletion(k8sAPI, options.namespace)
```

Comment:
are the args parsed for flags before calling ValidArgsFunction?  in my testing it seemed like it wasn't respecting the namespace flag.  We may need to explicitly parse args here to get the namespace flag.

Reply:
I figured we didn't need to parse flags based of this statement from the cobra [docs](https://github.com/spf13/cobra/blob/master/shell_completions.md#customizing-completions):
> Note: When using the ValidArgsFunction, Cobra will call your registered function after having parsed all flags and arguments provided in the command-line. You therefore don't need to do this parsing yourself. For example, when a user calls helm status --namespace my-rook-ns [tab][tab], Cobra will call your registered ValidArgsFunction after having parsed the --namespace flag, as it would have done when calling the RunE function.

It could be that that might not be true and we may indeed need to parse flags. How did you test this scenario?

Reply:
Ah, I think I was probably testing the wrong binary yesterday.  I gave this another try and the namespace flag worked perfectly: specifying `-n linkerd` showed the pods in the linkerd namespace, for example.  However, the `-A` flag did not seem to work and did not show pods from all namespaces.

Reply:
Ah ok. I'll fix the issue with the `-A` flag. good catch!